zero/src/rpc/native/txn.rs

```diff
@@ -55,37 +61,67 @@ where
 
 /// Processes the transaction with the given transaction hash and updates the
 /// accounts state.
-async fn process_transaction<ProviderT, TransportT>(
+pub async fn process_transaction<ProviderT, TransportT>(
     provider: &ProviderT,
     tx: &Transaction,
+    jumpdest_src: JumpdestSrc,
 ) -> anyhow::Result<(CodeDb, TxnInfo)>
 where
     ProviderT: Provider<TransportT>,
     TransportT: Transport + Clone,
 {
-    let (tx_receipt, pre_trace, diff_trace) = fetch_tx_data(provider, &tx.hash).await?;
+    let (tx_receipt, pre_trace, diff_trace, structlog_opt) =
+        fetch_tx_data(provider, &tx.hash, jumpdest_src).await?;
     let tx_status = tx_receipt.status();
     let tx_receipt = tx_receipt.map_inner(rlp::map_receipt_envelope);
     let access_list = parse_access_list(tx.access_list.as_ref());
 
-    let tx_meta = TxnMeta {
-        byte_code: <Ethereum as Network>::TxEnvelope::try_from(tx.clone())?.encoded_2718(),
-        new_receipt_trie_node_byte: alloy::rlp::encode(tx_receipt.inner),
-        gas_used: tx_receipt.gas_used as u64,
-    };
-
-    let (code_db, mut tx_traces) = match (pre_trace, diff_trace) {
+    let (mut code_db, mut tx_traces) = match (pre_trace, diff_trace) {
         (
             GethTrace::PreStateTracer(PreStateFrame::Default(read)),
             GethTrace::PreStateTracer(PreStateFrame::Diff(diff)),
-        ) => process_tx_traces(access_list, read, diff).await?,
+        ) => {
+            info!("{:?} {:?} {:?}", tx.hash, read, diff);
+            process_tx_traces(access_list, read, diff).await?
+        }
         _ => unreachable!(),
     };
 
     // Handle case when transaction failed and a contract creation was reverted
     if !tx_status && tx_receipt.contract_address.is_some() {
         tx_traces.insert(tx_receipt.contract_address.unwrap(), TxnTrace::default());
-    }
+    };
+
+    let jc: Option<(JumpDestTableWitness, CodeDb)> = structlog_opt.and_then(|struct_logs| {
```

Comment:
Is the `Option::None` case relevant to the user?
Can we keep it as just a bare collection?

That is, I turned a lot of `Option<HashMap<..>>` and `Option::unwrap_or_default` into bare collections in `trace_decoder` because they're not really needed - can you do the same here, or does something being absent have relevance?

Reply:
> Is the Option::None case relevant to the user?

Yes it is. We match this with the transaction list in the block. You can fail to deserialize/generate one transaction jumpdest table, but it could work for the others in the block. Doesn't mean that if we fail with one, we should abort the process.